lib/src/source/git.dart

```diff
@@ -435,16 +435,30 @@ class BoundGitSource extends CachedSource {
     _updatedRepos.add(path);
   }
 
-  ///  Clean-up [dirPath] if it's an invalid git repository
-  void _cleanInvalidGitRepoCache(String dirPath) {
-    if (dirExists(dirPath)) {
-      var processResult = runProcessSync(
-          git.command, ['rev-parse', '--is-inside-git-dir'],
-          workingDir: dirPath);
-      var result = processResult.stdout?.join('\n');
-      if (processResult.exitCode != 0 || result != 'true') {
-        deleteEntry(dirPath);
+  /// Clean-up [dirPath] if it's an invalid git repository.
+  ///
+  /// The git clones in the `PUB_CACHE` folder should never by invalid. But this
+  /// can happen if the clone operation failed in some way, and the program did
+  /// not exit gracefully, leaving the cache git clone in a dirty state.
+  Future<void> _cleanInvalidGitRepoCache(String dirPath) async {
```

Comment:
Maybe rename to `_deleteGitRepoIfInvalid` The current name is slightly confusing